### PR TITLE
[ADD]open_academy: Added new models Courses and Sessions with menu vi…

### PR DESCRIPTION
**Pull request by UlisesRivadeneyra**
-Add new odels:openacademy_sessions and openacademy_course
-Add the vies of the models added.
-Add menus of models added
-Add ir.model.access.csv
-Deleted files and docs __pycache__ T60932
-This PR include Alter existing content excersice

### CODE REVIEW

File: open_academy/models/course.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+
+class Course(models.Model):
+    _name = "openacademy.course"
+    _description = "Courses"
+
+    title = fields.Char(string="Courses", required=True)
+    description = fields.Text()
+
+    responsible_user = fields.Many2one(comodel_name="res.users", ondelete="set null")
```
Comment (edy1192) [inline open_academy/models/course.py:11]: You don't need to add comodel_name

```suggestion
    responsible_user = fields.Many2one("res.users", ondelete="set null")
```

File: open_academy/models/course.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+
+class Session(models.Model):
+    _name = "openacademy.sessions"
```
Comment (edy1192) [inline open_academy/models/course.py:15]: Use name in singular form please https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#37models

```suggestion
    _name = "openacademy.session"
```

File: open_academy/models/course.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    _description = "Sessions"
```
Comment (edy1192) [inline open_academy/models/course.py:16]:
```suggestion
    _description = "Session"
```

File: open_academy/models/course.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+    name = fields.Char(string="Sessions", required=True)
```
Comment (edy1192) [inline open_academy/models/course.py:18]:
```suggestion
    name = fields.Char(string="Session", required=True)
```

File: open_academy/models/course.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    start_date = fields.Date()
+    duration = fields.Float()
+    seats = fields.Integer(string="Number of seats")
+    instructor = fields.Many2one(comodel_name='res.partner', ondelete="set null")
+    course_id = fields.Many2one(comodel_name='openacademy.course', ondelete='')
```
Comment (edy1192) [inline open_academy/models/course.py:23]:
```suggestion
    instructor = fields.Many2one('res.partner', ondelete="set null")
    course_id = fields.Many2one('openacademy.course', ondelete='')
```

File: open_academy/views/course_menu_views.xml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <menuitem id="open_academy_course_menu" name="Open Academy Courses" action="openacademy_course_action_list"/>
+    <menuitem id="open_academy_session_menu" name="Open Academy Sessions" action="session_list_action"/>
```
Comment (edy1192) [inline open_academy/views/course_menu_views.xml:4]: You need a root menu before others menus.

```suggestion
    <menuitem id="open_academy_root_menu" name="Open Academy" sequence="1" />
    <menuitem id="open_academy_course_menu" name="Courses" sequence="1" parent="open_academy_root_menu" action="openacademy_course_action_list"/>
    <menuitem id="open_academy_session_menu" name="Sessions" sequence="2" parent="open_academy_root_menu" action="session_list_action"/>
```

File: open_academy/views/course_views.xml [hunk 7]
```diff
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (edy1192) [inline open_academy/views/course_views.xml:1]: Please move all session's views to another file `open_academy_session_views.xml`

File: open_academy/views/course_views.xml [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+<odoo>
+    <record id="openacademy_course_action_list" model="ir.actions.act_window">
+        <field name="name">List all courses</field>
```
Comment (edy1192) [inline open_academy/views/course_views.xml:4]:
```suggestion
        <field name="name">open.academy.course.action.list</field>
```

File: open_academy/views/course_views.xml [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+        <field name="res_model">openacademy.course</field>
+        <field name="view_mode">tree,form</field>
+    </record>
+    
+    <record id="session_list_action" model="ir.actions.act_window">
+        <field name="name">Sessions</field>
```
Comment (edy1192) [inline open_academy/views/course_views.xml:10]: This is the name of the action
```suggestion
        <field name="name">session.list.action</field>
```

File: open_academy/views/course_views.xml [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+        <field name="res_model">openacademy.sessions</field>
```
Comment (edy1192) [inline open_academy/views/course_views.xml:11]:
```suggestion
        <field name="res_model">openacademy.session</field>
```

File: open_academy/views/course_views.xml [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+            </tree>
+        </field>
+    </record> 
+
+    <!-- This is the search bar view of course-->
+    <record id="openacademy_course_search" model="ir.ui.view">
+        <field name="name">openacademy.course.view.search</field>
+        <field name="model">openacademy.course</field>
+        <field name="arch" type="xml">
+            <search string="">
```
Comment (edy1192) [inline open_academy/views/course_views.xml:52]: Remove string attribute please
```suggestion
            <search>
```

File: open_academy/views/course_views.xml [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+                <field name="title"></field>
+                <field name="description"></field>
+            </search>
+        </field>
+    </record> 
+
+    <!-- This is the tree view of course-->
+    <record id="openacademy_sessions_form" model="ir.ui.view">
```
Comment (edy1192) [inline open_academy/views/course_views.xml:60]:
```suggestion
    <record id="openacademy_session_form" model="ir.ui.view">
```

File: open_academy/views/course_views.xml [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+        <field name="name">openacademy.sessions.form</field>
```
Comment (edy1192) [inline open_academy/views/course_views.xml:61]:
```suggestion
        <field name="name">openacademy.session.form</field>
```

File: open_academy/views/course_views.xml [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+        <field name="model">openacademy.sessions</field>
```
Comment (edy1192) [inline open_academy/views/course_views.xml:62]:
```suggestion
        <field name="model">openacademy.session</field>
```

File: open_academy/views/course_views.xml [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+        <field name="model">openacademy.course</field>
+        <field name="arch" type="xml">
+            <search string="">
+                <field name="title"></field>
+                <field name="description"></field>
+            </search>
+        </field>
+    </record> 
+
+    <!-- This is the tree view of course-->
```
Comment (edy1192) [inline open_academy/views/course_views.xml:59]:
```suggestion
    <!-- This is the tree view of session-->
```

And where is the session's tree view?

File: open_academy/README.rst [hunk 16]
```diff
@@ -0,0 +1,35 @@
+**This file is going to be generated by oca-gen-addon-readme.**
```
Comment (edy1192) [inline open_academy/README.rst:1]: Please remove all this and add a description for this module.

**Review by edy1192 — CHANGES_REQUESTED** [review]

**Comment by edy1192** [comment]
@UlisesRivadeneyra please add a description for your PR

File: open_academy/views/course_views.xml (hunk 7 above)
Comment (UlisesRivadeneyra) [reply open_academy/views/course_views.xml]: HERE

**Comment by UlisesRivadeneyra** [comment]
Hey @edy1192 Im ready to check my second PR, thanks for your time, greetings!!!

File: .flake8 [hunk 17]
```diff
@@ -0,0 +1,12 @@
+[flake8]
```
Comment (edy1192) [inline .flake8:1]: Please remove this file and add it to gitignore file.

File: open_academy/demo/open_academy_demo.xml [hunk 18]
```diff
@@ -0,0 +1,17 @@
...
+            <field name="title">XML course</field>
+            <field name="description">This is the XML course</field>
+        </record>
+        <record id="psl_course" model="openacademy.course">
+            <field name="title">PSQL Course</field>
+            <field name="description">This is the PSQL course</field>
+        </record>
+</odoo>
+
+
```
Comment (edy1192) [inline open_academy/demo/open_academy_demo.xml:17]: Remove blank lines please

File: open_academy/__manifest__.py [hunk 19]
```diff
@@ -0,0 +1,27 @@
...
+    'category': 'Uncategorized',
+    'version': '15.0.1.0.0',
+    # any module necessary for this one to work correctly
+    'depends': ['base'],
+    # always loaded
+    'data': [
+        'security/ir.model.access.csv',
+        'views/open_academy_course_views.xml',
+        'views/open_academy_menu_views.xml',
+        'views/open_academy_session_views.xml'
```
Comment (edy1192) [inline open_academy/__manifest__.py:21]: This goes after views.
```suggestion
        'views/open_academy_session_views.xml',
        'views/open_academy_menu_views.xml'
```

File: open_academy/models/open_academy_course.py [hunk 20]
```diff
@@ -0,0 +1,13 @@
...
+class Course(models.Model):
+    _name = "openacademy.course"
+    _description = "Course"
+
+    title = fields.Char(string="Courses", required=True)
+    description = fields.Text()
+
+    responsible_user = fields.Many2one("res.users", ondelete="set null")
+
+
```
Comment (edy1192) [inline open_academy/models/open_academy_course.py:13]: Remove blank lines please.

File: open_academy/models/open_academy_course.py [hunk 21]
```diff
@@ -0,0 +1,13 @@
+from odoo import models, fields
+
+
+class Course(models.Model):
+    _name = "openacademy.course"
+    _description = "Course"
+
+    title = fields.Char(string="Courses", required=True)
```
Comment (edy1192) [inline open_academy/models/open_academy_course.py:8]:
```suggestion
    title = fields.Char(string="Course", required=True)
```

File: open_academy/models/open_academy_session.py [hunk 22]
```diff
@@ -0,0 +1,13 @@
+from odoo import models, fields
+
+
+class Session(models.Model):
+    _name = "openacademy.session"
+    _description = "Session"
+
+    name = fields.Char(string="Sessions", required=True)
```
Comment (edy1192) [inline open_academy/models/open_academy_session.py:8]:
```suggestion
    name = fields.Char(string="Session", required=True)
```

File: open_academy/models/open_academy_session.py [hunk 23]
```diff
@@ -0,0 +1,13 @@
...
+    start_date = fields.Date()
+    duration = fields.Float()
+    seats = fields.Integer(string="Number of seats")
+    instructor = fields.Many2one('res.partner', ondelete="set null")
```
Comment (edy1192) [inline open_academy/models/open_academy_session.py:12]: See the guidelines https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#38fields
```suggestion
    instructor_id = fields.Many2one('res.partner', ondelete="set null")
```

File: open_academy/views/open_academy_course_views.xml [hunk 24]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="openacademy_course_action_list" model="ir.actions.act_window">
+        <field name="name">open.academy.course.action.list</field>
```
Comment (edy1192) [inline open_academy/views/open_academy_course_views.xml:4]: Sorry, I got confused with this. I thought it was the view and it is the action. This is what the user sees when he opens the view, so please change the name.

```suggestion
        <field name="name">Courses</field>
```
Sorry 🙊

File: open_academy/views/open_academy_course_views.xml [hunk 25]
```diff
@@ -0,0 +1,59 @@
...
+        <field name="res_model">openacademy.course</field>
+        <field name="view_mode">tree,form</field>
+    </record>
+    
+    <record id="session_list_action" model="ir.actions.act_window">
+        <field name="name">session.list.action</field>
+        <field name="res_model">openacademy.session</field>
+        <field name="view_mode">tree,form</field>
+    </record>
```
Comment (edy1192) [inline open_academy/views/open_academy_course_views.xml:13]: Move this to `open_academy_session_views.xml` file please.

And sorry, I got confused with this. I thought it was the view and it is the action. This is what the user sees when he opens the view, so please change the name.

```suggestion
        <field name="name">Sessions</field>
```
Sorry 🙊

File: open_academy/views/open_academy_session_views.xml [hunk 26]
```diff
@@ -0,0 +1,22 @@
...
+        <field name="model">openacademy.session</field>
+        <field name="arch" type="xml">
+            <form string="Course Sesion">
+                <sheet>
+                    <group>
+                        <field name="name"/>
+                        <field name="start_date"/>
+                        <field name="duration"/>
+                        <field name="seats"/>
+                    </group>
```
Comment (edy1192) [inline open_academy/views/open_academy_session_views.xml:16]: Where are `course_id` and `instructor_id` fields?

**Review by edy1192 — CHANGES_REQUESTED** [review]

**Comment by UlisesRivadeneyra** [comment]
@edy1192 Hi, i already change the comments that you did, my PR is ready for the new review, thanks!! Greetings, Ulises!!

File: open_academy/models/open_academy_course.py [hunk 27]
```diff
@@ -0,0 +1,11 @@
...
+
+
+class Course(models.Model):
+    _name = "openacademy.course"
+    _description = "Course"
+
+    title = fields.Char(string="Course", required=True)
+    description = fields.Text()
+
+    responsible_user = fields.Many2one("res.users", ondelete="set null")
```
Comment (edy1192) [inline open_academy/models/open_academy_course.py:11]: In anothe PR please change this field. Follow the naming guidelines for Many2One fields

```suggestion
    responsible_user_id = fields.Many2one("res.users", ondelete="set null")
```

**Review by edy1192 — APPROVED** [review]

**Comment by edy1192** [comment]
Merged @UlisesRivadeneyra 👏🏼

**Pull request merged**